### ISSUE TITLE
Create basic samples dashboard [8]

### ISSUE THREAD

**Issue by KristinaBorch** [hzi-braunschweig/SORMAS-Project#10721]
### Feature Description

For the creation of a sample dashboard, a new button to switch between the surveillance, contact and sample dashboard is needed. 
The creation of sample dashboard is tracked in https://github.com/hzi-braunschweig/SORMAS-Project/issues/10723. 

### Proposed Change

- [x] Add a `DASHBOARD_SAMPLES_VIEW` user right and give it to all default user roles with the `DASHBOARD_SURVEILLANCE_VIEW` and `SAMPLES_VIEW` user rights.
- [x] Extend the dashboard switcher option group with a "Sample" option.
![Enum boxes fields](https://user-images.githubusercontent.com/98821699/197716062-72fea509-4e60-42e0-b30f-45ae1b347028.png)
- [x] Add a feature type property called `DASHBOARD_SAMPLES` to the `DASHBOARD` feature type, default to true. If set to false, remove the "Sample" option from the dashboard switcher
  - [x] Ideally (TBD whether this is out of scope of this issue, but for consistency reasons I @MateStrysewske think it makes sense), also add `DASHBOARD_SURVEILLANCE`, `DASHBOARD_CONTACTS` and `DASHBOARD_CAMPAIGNS` feature type properties and apply the same logic to the dashboard switcher; if all three feature types are set to false, don't display the dashboard at all
- [x] Add filters: In general look at all samples the user can access. Exclude deleted samples, include archived ones.
  - region of the associated entity. Note that a sample may have multiple associated entities, so duplicates need to be excluded. Add a tooltip to the filter
  - district. Same as region
  - type of sample (sampleMaterial)
  - date filter allowing to filter by date or epi week, as on surveillance dashboard, with the following options:
    1. Sample collection date (sampleDateTime)
    2. Report date of associated entity (e.g. case, contact, event of event participant)
    3. Most Relevant date: priority is pathogen test date (earliest if multiple exist for the sample) + fall back to sample collection date. Associated entity is not relevant
  - disease (of the associated entity); including the option to choose "no disease" (NOT similar to an empty filter!), which is necessary for event participants of events without an assigned disease -> use a dropdown filter as on the contact dashboard
- [x] Add an element that displays the overall samples count as in the following picture.  
           - All samples == total number of samples,
           - the counts and proportions correspond to the variable  "final lab results": Add counts and proportions for `Indeterminate, pending, negative, positive, not done`.  This is a bit different from the widget on the surveillance dashboard, because it considers all samples (not only the latest sample of cases) and also shows the "not done" status
![image](https://user-images.githubusercontent.com/12996724/215931749-5469a643-b846-4c87-a9f2-5845aeb972e8.png)

**OPEN QUESTIONS:**
- What should the region and district filters filter by? The jurisdiction of their parent entity (e.g. case, contact) or the address of the associated person (because that's what is supposed to be used on the map according to #10745)? --> Proposal by @MartinWahnschaffe Let's go with the jurisdiction of the parent entity. The person's address may be empty and from the responsibility point of view the jurisdiction is the right choice, I think.
- ~~How do we deal with "copied" samples? E.g. when we create a resulting case from a contact, the sample of the contact is duplicated, so in SORMAS, we now have two samples (one for the contact, one for the case) referring to the same physical sample. If we count those, we end up with a higher sample number than the number of samples that have been physically taken; if we don't count those, how do we determine which sample to pull the final laboratory result from?~~ Normally case and contact share the same sample.

### Acceptance Criteria

* The button Sample is implemented as described above.
* sum of count for all enums == All samples
* sum of proportion ==100%
* The needed database query may be slow. Should be tested on a system with at least 10k samples and the resulting query (with region filter active) should be documented with https://explain.dalibo.com/.

**Comment by MartinWahnschaffe** [comment]
Merged into #10732 and #10724 to have a minimal working functionality.

**Comment by MateStrysewske** [comment]
@bernardsilenou

There are two open questions that we need to clarify before starting development on this issue:

- What should the region and district filters filter by? The jurisdiction of their parent entity (e.g. case, contact) or the address of the associated person (because that's what is supposed to be used on the map according to https://github.com/hzi-braunschweig/SORMAS-Project/issues/10745)?
- How do we deal with "copied" samples? E.g. when we create a resulting case from a contact, the sample of the contact is duplicated, so in SORMAS, we now have two samples (one for the contact, one for the case) referring to the same physical sample. If we count those, we end up with a higher sample number than the number of samples that have been physically taken; if we don't count those, how do we determine which sample to pull the final laboratory result from?

**Comment by MartinWahnschaffe** [comment]
> * How do we deal with "copied" samples? E.g. when we create a resulting case from a contact, the sample of the contact is duplicated, so in SORMAS, we now have two samples (one for the contact, one for the case) referring to the same physical sample. If we count those, we end up with a higher sample number than the number of samples that have been physically taken; if we don't count those, how do we determine which sample to pull the final laboratory result from?

From what I remember and see, when I check the source code, the sample is only cloned when the resulting case has a different disease than the contact or when the sample is already associated to another case. Normally the sample is shared by contact and case (see CaseFacadeEjb.setSampleAssociations).

**Comment by MartinWahnschaffe** [comment]
> * What should the region and district filters filter by? The jurisdiction of their parent entity (e.g. case, contact) or the address of the associated person (because that's what is supposed to be used on the map according to

Let's go with the jurisdiction of the parent entity. The person's address may be empty and from the responsibility point of view the jurisdiction is the right choice, I think.

**Comment by bernardsilenou** [comment]
@MartinWahnschaffe  Following  this comment https://github.com/hzi-braunschweig/SORMAS-Project/issues/10721#issuecomment-1432794429
- We had the idea to use the person's address when showing nodes on the map, issue https://github.com/hzi-braunschweig/SORMAS-Project/issues/10745.  It can happen that the responsible jurisdiction used in global filer is Region1 while the person's home address is in Region2, resulting in the icon being placed in Region2
- I'd like to know if we should add the option to view by home address or responsible jurisdiction (or facility if applicable) to #10745
- A similar situation applies to the map on the contact dashboard, when we click on "show contacts", which data is used here?

**Comment by bernardsilenou** [comment]
Following https://github.com/hzi-braunschweig/SORMAS-Project/issues/10721#issuecomment-1432777000
if a sample is cloned, then we can consider them  as independent samples even though they  have exact attributes and linked to the same person

**Comment by MartinWahnschaffe** [comment]
> @MartinWahnschaffe Following this comment [#10721 (comment)](https://github.com/hzi-braunschweig/SORMAS-Project/issues/10721#issuecomment-1432794429)
> 
>     * We had the idea to use the person's address when showing nodes on the map, issue [Add sample status map to sample dashboard #10745](https://github.com/hzi-braunschweig/SORMAS-Project/issues/10745).  It can happen that the responsible jurisdiction used in global filer is Region1 while the person's home address is in Region2, resulting in the icon being placed in Region2

Guess that's ok then, right?

>     * I'd like to know if we should add the option to view by home address or responsible jurisdiction (or facility if applicable) to [Add sample status map to sample dashboard #10745](https://github.com/hzi-braunschweig/SORMAS-Project/issues/10745)
> 
>     * A similar situation applies to the map on the contact dashboard, when we click on "show contacts", which data is used here?

I have replied here: https://github.com/hzi-braunschweig/SORMAS-Project/issues/10745#issuecomment-1434379087

**Comment by MartinWahnschaffe** [comment]
@bernardsilenou Another question: Let's say a case is associated to both a case and a contact. We are using the "Report date of associated entity" as filter. The contact report date is in week 1, the case report date in week 2. Should the sample be shown in the dashboard when we filter for week 1 and also when we filter for week 2?

**Comment by bernardsilenou** [comment]
@MartinWahnschaffe 
- Yes please, the sample should show for both weeks.  Suggestion:

1.    If date interval is week1, then we count sample + use contact widget on map
2.    If date interval is week2, then we count sample + use case widget on map
3.    If date interval covers week 2 and week1, then we count sample (only once) + use case widget on map (ie use the parent entity node with most recent report date)

**Comment by leventegal-she** [comment]
Analysis of the query used to get the counts https://explain.dalibo.com/plan/71fe5dd39460c7b7#plan

**Comment by adinaflorea9** [comment]
Reopening ticket so that the heading can be updated:
![image](https://user-images.githubusercontent.com/96909231/222741977-84c1ac5d-d686-42c3-ad9b-9d063a871802.png)

**Comment by bernardsilenou** [comment]
@MateStrysewske @adinaflorea9 
I had a look at the implementation and its very nice. Some minor editing suggestions:

1. Rename Reporting date of associated entity to Report date of associated entity
2. Sort enums of date filter with most relevant date at top since its the default, as on surveillance dashboard
3. Sort enums for type of sample alphabetically using users language (because of translation)
4. Capitalised first letters in captions for "Final Laboratory Results" "All Samples", "No disease" (enum for disease filter)
I do not know if all can be done in this ticket.

**Comment by leventegal-she** [comment]
Repoen to add warning that currently all samples from the system regredless of the user jurisdiction. A followup ticket will be created to filter the samples by user jusridiction.

**Comment by adinaflorea9** [comment]
Verified ticket on the local machine using the latest version of SORMAS from the development branch - 1.81.0-SNAPSHOT(156c4e6).

A follow-up issue was created to address the fact that the dashboard currently displays all samples regardless of the user's jurisdiction - #11605 

An informative note was added to let the user know about the results displayed:
![image](https://user-images.githubusercontent.com/96909231/223697205-4e8ff651-79a8-4585-a885-643e0f947897.png)
